### PR TITLE
Labels feature

### PR DESCRIPTION
**Pull request by nofaralfasi**
Added functionality for the label feature

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
now after #3 got merged,
this PR deserves a good rebase :)

File: .gitignore
Comment (Ron-Lavi) [inline .gitignore:40]: Interesting, it's already part of the updated main branch, did you fetch the new changes? :)

**Comment by nofaralfasi** [comment]
Please do not check yet this PR. Thanks

File: .gitignore
Comment (Ron-Lavi) [inline .gitignore:42]: In case there are more files that are generated by the IDE
```suggestion
/.idea/*
```

File: app/controllers/labels_controller.rb
Comment (Ron-Lavi) [inline app/controllers/labels_controller.rb:2]: since it's only applicable for the `destroy` action,
should we move it into the action itself? e.g:
```ruby
def destroy
    @label = Label.find_by(id: params[:id])

    if @label.blank?
      redirect_to root_path, notice: "Label #{params[:id]} was not found."
    end
    
    if @label.destroy
        redirect_to root_path, notice: "Label `#{@label.title}` was successfully deleted."
    end
end
```

File: app/controllers/labels_controller.rb
Comment (Ron-Lavi) [inline app/controllers/labels_controller.rb:23]: should it be: ?
```suggestion
    if @label.destroy
        redirect_to root_path, notice: "Label `#{@label.title}` was successfully deleted."
    end
```

File: app/controllers/tasks_controller.rb
Comment (Ron-Lavi) [inline app/controllers/tasks_controller.rb:13]: I see `@labels = Label.all`,
should we move it into a `before_action` callback?

File: app/models/task.rb
Comment (Ron-Lavi) [inline app/models/task.rb:3]: IMHO each task `has_one :label`

**Review by Ron-Lavi — COMMENTED** [review]
Works great @nofaralfasi !
left a few comments :)

File: app/models/task.rb
Comment (nofaralfasi) [reply app/models/task.rb:3]: According to my understanding, the `has_one` association specifies a one-to-one association with another class, but in our case we need a one-to-many association with the label class.
However I made a mistake writing `labels` (in plural) instead of `label`, so I will fix it.

**Review by Ron-Lavi — APPROVED** [review]
Thanks @nofaralfasi, LGTM 👍🏼 
leaving this one to be merged by @stejskalleos :)

File: app/views/lists/show.html.erb
Comment (stejskalleos) [inline app/views/lists/show.html.erb:18]: Try to keep `views` slim and don't put much logic in them. For this case you should have helper method `task_status` that would contain logic for what should be displayed and what not.

See https://www.rubyguides.com/2020/01/rails-helpers/

File: app/views/tasks/show.html.erb
Comment (stejskalleos) [reply app/views/tasks/show.html.erb:18]: And use helper method here, as per my comment above

File: app/views/lists/show.html.erb
Comment (stejskalleos) [inline app/views/lists/show.html.erb:31]: don't use `:class =>` but `class:`

File: app/views/lists/index.html.erb
Comment (stejskalleos) [inline app/views/lists/index.html.erb:26]: don't use `:class =>` but `class:`

File: app/views/tasks/_form.html.erb
Comment (stejskalleos) [inline app/views/tasks/_form.html.erb:28]: use `if.any?` instead of `unless.empty?`

**Review by stejskalleos — CHANGES_REQUESTED** [review]

**Comment by nofaralfasi** [comment]
Hi @stejskalleos, 
I am aware that there are a lot of syntax issues I need to fix (like writing `:class =>`, instead of `class:`), and I am planning to fix them on a separate commit (maybe with the use of rubocop).

File: .ruby-version
Comment (stejskalleos) [inline .ruby-version:1]: TIL that you can't set ruby version in `.ruby-version` file without the patch version, eg `ruby-2.7`.
Because of this my local dev environment doesn't work because my ruby version is `2.7.4`.

In ideal case we should use only `Gemfile` for specifying the version like this `ruby "~> 2.7.0"`

Here is great article about it: https://andycroll.com/ruby/use-loose-ruby-versioning-in-your-gemfile/

File: app/helpers/tasks_helper.rb
Comment (stejskalleos) [inline app/helpers/tasks_helper.rb:2]: NIT:
If you have instance variable in controller for example `@task`, you don't have to pass task as argument but you can do this:
```ruby
# In helper:
  def task_status
    @task.status ? 'Done' : 'To Do'
  end

  def task_label
    @task.label&.title || 'None'
  end

# In views:
<%= task_status %>
<%= task_label %>
```

File: app/models/label.rb
Comment (stejskalleos) [inline app/models/label.rb:4]: NIT: Validation for unique name would be also handy

**Review by stejskalleos — APPROVED** [review]
:heavy_check_mark: **LGTM**
I have few comments but all of them are more of NITs and TIL details that doesn't have to block the PR.
Merging!

**Comment by Ron-Lavi** [comment]
Thanks @stejskalleos !!

**Pull request merged**
